File: 0900-reordered-power-of-2/0900-reordered-power-of-2.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        s = str(n)
        s = "".join(sorted(s))
        i = 1
        while i < (1 << 31) - 1:
            cmp = str(i)
            cmp = "".join(sorted(cmp))
            if cmp == s:
                return True
            i <<= 1
        return False

File: 0900-reordered-power-of-2/0900-reordered-power-of-2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        string s = to_string(n);
        sort(s.begin(), s.end());
        for(unsigned int i = 1; i < (1ul << 31) - 1; i <<= 1)
        {
            string cmp = to_string(i);
            sort(cmp.begin(), cmp.end());
            if(cmp == s)
            {
                return true;
            }
        }
        return false;
    }
};